src/utils/api.ts

```diff
@@ -33,6 +34,22 @@ export class ApiService {
     this.app.get("/", (_req: Request, res: Response) => {
       res.send("🐮 Moooo!");
     });
+    this.app.get("/health", async (_req: Request, res: Response) => {
+      // Using an iterator, process all the chain contexts, storing the health
+      // in a map, and if any of the contexts are unhealthy, return false
+      const contexts = Object.values(ChainContext.chains) as ChainContext[];
+      const healths = contexts.map((context) => context.health);
+      const healthStatusDict = healths.reduce((acc, health) => {
+        acc[health.chainId] = health;
+        return acc;
+      }, {} as { [chainId: string]: ChainHealth });
+
+      const overallHealth = healths.every((health) => health.isHealthy);
+
+      res
+        .status(overallHealth ? 200 : 500)
+        .send({ overallHealth, ...healthStatusDict });
```

Comment:
The logic, looks good to me, but for `readyness`, I would implement `liveness` based on the same criteria of watch dog (and remove watch dog --> or allow to disable it by config)

Reply:
Readiness most probably isn't appropriate (see above comments). Liveness does use the same criteria as the watch dog, with the watch now setting sync status to `UNKNOWN` if it detects the timeout (and it's _not_ running in a kubernetes environment).